### DISCUSSION TITLE
Does SqlClient support provider related settings added in appsettings.json file for ASP.NET core apps

### DISCUSSION THREAD

**Discussion by Trivedi-Ashish** [dotnet/SqlClient#1571] [Q&A] [answered]
I am planning to use SqlClient library for my ASP.NET core sample app.
ASP.NET core comes with appsettings.json where one can add all configuration settings related to database provider as well.
So, does SqlClient supports that?
If yes, what all settings I may add in that, also the specific keys to which I will be adding my values to?

Let me know if this clear or I need to be more specific?

**Comment by Kaur-Parminder** [comment, marked as answer]
what all settings I may add in that, also the specific keys to which I will be adding my values to?

@Trivedi-Ashish Yes, you can use appsettings.json in ASP.NetCore Application for using Microsoft.Data.SqlClient features. 

The settings depends upon feature and design you will be using in application, just as for opening a connection you can set connection string as a setting. You may choose any feature setup requirements to be  configured in appsettings.json.

**Reply by Trivedi-Ashish** [reply]
@Kaur-Parminder 
Are those settings configured in appsettings.json read by SqlClient directly -or- developer needs to read those settings inside their ASP.NET core app implementation and feed it to the SqlClient while creating SqlConnection instance?

I am assuming from your answer that SqlClient doesn't read appsettings.json directly, instead the application is supposed to read desired settings from appsettings.json and pass them to SqlClient. Is that correct?

**Reply by Kaur-Parminder** [reply]
I am assuming from your answer that SqlClient doesn't read appsettings.json directly, instead the application is supposed to read desired settings from appsettings.json and pass them to SqlClient. Is that correct?

That is correct.

**Reply by Trivedi-Ashish** [reply]
@Kaur-Parminder Hello, I know that this discussion is answered and closed. But I do have one more query related to same topic.
I wanted to know if there are any configuration files (.xml/.json/etc) which are read by sqlClient directly?
If yes, are those internal to SqlClient or can app developers modify those files as per their needs?
